Realm/RLMRealm.mm

```diff
@@ -693,16 +693,17 @@ + (uint64_t)schemaVersionAtURL:(NSURL *)fileURL encryptionKey:(NSData *)key erro
     }
 }
 
-+ (NSError *)migrateRealm:(RLMRealmConfiguration *)configuration {
++ (BOOL)migrateRealm:(RLMRealmConfiguration *)configuration error:(NSError **)error {
+
     if (RLMGetAnyCachedRealmForPath(configuration.config.path)) {
         @throw RLMException(@"Cannot migrate Realms that are already open.");
     }
 
     @autoreleasepool {
-        NSError *error = nil;
-        [RLMRealm realmWithConfiguration:configuration error:&error];
-        return error;
+        [RLMRealm realmWithConfiguration:configuration error:error];
+        if (error) return NO;
```

Comment:
iirc, the value of an error parameter should not be checked unless the functions returns `nil` or `NO`